sklearn/preprocessing/label.py

```diff
@@ -891,14 +894,21 @@ def transform(self, y):
         """
         check_is_fitted(self, 'classes_')
 
-        class_to_index = dict(zip(self.classes_, range(len(self.classes_))))
+        class_to_index = self._cached_dict_for()
         yt = self._transform(y, class_to_index)
 
         if not self.sparse_output:
             yt = yt.toarray()
 
         return yt
 
+    def _cached_dict_for(self):
```

Comment:
the method should be named as a verb, e.g. `_build_cache`

Reply:
renamed